collect_app/src/androidTest/java/org/odk/collect/android/regression/FillBlankFormTest.java

```diff
@@ -67,108 +68,114 @@
     public void subtext_ShouldDisplayAdditionalInformation() {
 
         //TestCase2
-        MainMenu.clickFillBlankForm();
-        MainMenu.checkIsFormSubtextDisplayed();
+        new MainMenuPage(main)
+                .clickFillBlankForm()
+                .checkIsFormSubtextDisplayed();
 
     }
 
     @Test
     public void exitDialog_ShouldDisplaySaveAndIgnoreOptions() {
 
         //TestCase6 , TestCase9
-        MainMenu.startBlankForm("All widgets");
-        pressBack();
-        FormEntry.checkIsStringDisplayed(R.string.keep_changes);
-        FormEntry.checkIsStringDisplayed(R.string.do_not_save);
-        FormEntry.clickOnString(R.string.do_not_save);
-        FormEntry.checkIsIdDisplayed(R.id.enter_data);
-        FormEntry.checkIsIdDisplayed(R.id.get_forms);
-
+        new MainMenuPage(main)
+                .startBlankForm("All widgets")
+                .pressBack(new ExitFormDialog("All widgets", main))
+                .checkIsStringDisplayed(R.string.keep_changes)
+                .checkIsStringDisplayed(R.string.do_not_save)
+                .clickOnString(R.string.do_not_save)
+                .checkIsIdDisplayed(R.id.enter_data)
+                .checkIsIdDisplayed(R.id.get_forms);
     }
 
     @Test
     public void searchBar_ShouldSearchForm() {
 
         //TestCase12
-        MainMenu.clickFillBlankForm();
-        MainMenu.clickMenuFilter();
-        MainMenu.searchInBar("Aaa");
-        pressBack();
-        pressBack();
-
+        new MainMenuPage(main)
+                .clickFillBlankForm()
+                .clickMenuFilter()
+                .searchInBar("Aaa")
+                .pressBack(new BlankFormSearchPage(main))
```

Comment:
We could also have each Page implement its own `pressBack` (similar to `assertOnaPage`) but in my experience where "back" is ends up being pretty dynamic and I also like how this ends up giving you extra documentation in the test around where you are.